components/Spaces/CallTabs/Music/Tabs/SearchSongs.js

```diff
@@ -39,29 +41,43 @@ const SearchSongs = () => {
     let cancel = false;
     spotifyApi.searchTracks(search).then((res) => {
       if (cancel) return;
-      setSearchResults(
-        res.body.tracks.items.map((track) => {
-          const smallestAlbumImage = track.album.images.reduce((smallest, image) => {
-            if (image.height < smallest.height) return image;
-            return smallest;
-          }, track.album.images[0]);
-
-          return {
-            artist: track.artists[0].name,
-            title: track.name,
-            uri: track.uri,
-            albumUrl: smallestAlbumImage.url,
-          };
-        })
-      );
+      setSearchResults(parseTracks(res.body.tracks.items));
     });
 
     return () => (cancel = true);
   }, [search, accessToken]);
 
   const playTrack = (track) => {
     setSearchResults([]);
-    setTrackUri(track.uri);
+    setQueue((prev) => {
+      let newQueue = [];
+      if (prev.length === 0) {
+        newQueue.push(track);
+      } else {
+        // spotify player has bug with setting a new queue if it's the same length.
+        // This is an inperfect workaround to make sure the song changes.
+        newQueue = [track, currentTrack, ...nextTracks];
+      }
+      return newQueue;
+    });
+  };
+
+  const addToQueue = (track) => {
+    let newQueue = [];
+    setQueue((prev) => {
+      newQueue = [...prev];
+      newQueue.push(track);
+      return newQueue;
+    });
+    if (newQueue.length <= 1) {
+      setCurrentTrack(track);
+    } else {
+      setNextTracks((prev) => {
+        const newNextTrack = [...prev];
+        newNextTrack.push(track);
+        return newNextTrack;
```

Comment:
cant u do `setNextTracks((prev) => [...prev, track])` ?

Reply:
That's actually big brain optimization 🚀

Reply:
yes hehe. You reminded me. I actually have to change that function lol.